nixos/modules/services/networking/supplicant.nix

```diff
@@ -1,42 +1,40 @@
 { config, lib, utils, pkgs, ... }:
-
-with lib;
-
 let
 
   cfg = config.networking.supplicant;
 
-  # We must escape interfaces due to the systemd interpretation
+  # We must lib.escape interfaces due to the systemd interpretation
   subsystemDevice = interface:
-    "sys-subsystem-net-devices-${utils.escapeSystemdPath interface}.device";
+    "sys-subsystem-net-devices-${utils.lib.escapeSystemdPath interface}.device";
 
   serviceName = iface: "supplicant-${if (iface=="WLAN") then "wlan@" else (
                                      if (iface=="LAN") then "lan@" else (
                                      if (iface=="DBUS") then "dbus"
-                                     else (replaceStrings [" "] ["-"] iface)))}";
+                                     else (lib.replaceStrings [" "] ["-"] iface)))}";
 
   # TODO: Use proper privilege separation for wpa_supplicant
   supplicantService = iface: suppl:
     let
       deps = (if (iface=="WLAN"||iface=="LAN") then ["sys-subsystem-net-devices-%i.device"] else (
              if (iface=="DBUS") then ["dbus.service"]
-             else (map subsystemDevice (splitString " " iface))))
-             ++ optional (suppl.bridge!="") (subsystemDevice suppl.bridge);
-
-      ifaceArg = concatStringsSep " -N " (map (i: "-i${i}") (splitString " " iface));
-      driverArg = optionalString (suppl.driver != null) "-D${suppl.driver}";
-      bridgeArg = optionalString (suppl.bridge!="") "-b${suppl.bridge}";
-      extraConfFile = pkgs.writeText "supplicant-extra-conf-${replaceStrings [" "] ["-"] iface}" ''
-        ${optionalString suppl.userControlled.enable "ctrl_interface=DIR=${suppl.userControlled.socketDir} GROUP=${suppl.userControlled.group}"}
-        ${optionalString suppl.configFile.writable "update_config=1"}
+             else (map subsystemDevice (lib.splitString " " iface))))
+             ++ lib.optional (suppl.bridge!="") (subsystemDevice suppl.bridge);
+
+      ifaceArg = lib.concatStringsSep " -N " (map (i: "-i${i}") (lib.splitString " " iface));
+      driverArg = lib.optionalString (suppl.driver != null) "-D${suppl.driver}";
+      bridgeArg = lib.optionalString (suppl.bridge!="") "-b${suppl.bridge}";
+      confFileArg = lib.optionalString (suppl.configFile.path!=null) "-c${suppl.configFile.path}";
+      extraConfFile = pkgs.writeText "supplicant-extra-conf-${lib.replaceStrings [" "] ["-"] iface}" ''
+        ${lib.optionalString suppl.userControlled.enable "ctrl_interface=DIR=${suppl.userControlled.socketDir} GROUP=${suppl.userControlled.group}"}
+        ${lib.optionalString suppl.configFile.writable "update_config=1"}
```

Comment:
There's an extra and unused `confFileArg`

Reply:
sorry. I am confused about this comment... is this still relevant?

Reply:
You fixed it with the recent force push, thanks

Just to clear the confusion: I was talking about the introduction of a new variable, `confFileArg`, which wasn't referenced anywhere.

![Untitled](https://github.com/user-attachments/assets/089cecb0-7ead-4b4c-b18b-501c992e3d5b)

Reply:
thanks for clarification. I guess i can close this thread.